src/main/java/domain/Frames.java

```diff
@@ -0,0 +1,44 @@
+package domain;
+
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class Frames {
+    public static final int MAX_FRAME_COUNT = 10;
+    private final List<Frame> frames;
+
+    private Frames() {
+        final List<Frame> frames = Stream.generate(NormalFrame::newInstance)
+                .limit(MAX_FRAME_COUNT - 1)
+                .collect(Collectors.toList());
+
+        frames.add(FinalFrame.newInstance());
+        this.frames = Collections.unmodifiableList(frames);
+    }
+
+    public static Frames newInstance() {
+        return new Frames();
+    }
+
+    public boolean isEnd() {
+        return this.frames.stream()
+                .allMatch(Frame::isDone);
+    }
+
+    public void throwBowlingBall(int inputFallenPins) {
+        final Frame proceedFrame = this.frames.stream()
+                .filter(frame -> !frame.isDone())
```

Comment:
filter() 메소드의 predicate 인자의 특성으로 메소드 레퍼런스를 이용하여 아래와 같이 표현하는것도 좋겠다는 생각이 들어요.
```suggestion
                .filter(frame::isProceed)
```

Reply:
오 가독성이 훨씬 좋아지겠네요! 고려해보겠습니다!